src/node_buffer.cc

```diff
@@ -550,6 +566,9 @@ void ByteLengthUtf8(const FunctionCallbackInfo<Value> &args) {
 
 
 void Compare(const FunctionCallbackInfo<Value> &args) {
+  CHECK_IS_BUFFER(Environment::GetCurrent(args), args[0]);
+  CHECK_IS_BUFFER(Environment::GetCurrent(args), args[1]);
```

Comment:
Did this because `Environment::GetCurrent()` is only called if a conditional fails. Will change.